File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        vector<int> one_dim_array;
        for(vector<int> nums: matrix){
            for(int num: nums){
                one_dim_array.push_back(num);
            }
        }
        




        int left = 0;
        int right = one_dim_array.size()-1;

        while(left <= right){
            int mid = left + (right-left)/2;
            if (one_dim_array[mid] == target){
                return true;
            }
            else if (one_dim_array[mid] < target){
                left = mid+1;
            }
            else{
                right = mid-1; 
            }
        }

        return false;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()

        solution = []

        for i, num in enumerate(nums):
            if num > 0:
                break
            elif i > 0 and num == nums[i-1]:
                continue
            j = i+1
            k = len(nums)-1
            while k > j:
                n1, n2 = nums[j], nums[k]
                if num + n1 + n2 == 0:
                    solution.append([num, n1, n2])
                    j +=1
                    k -=1
                    while nums[j] == nums[j-1] and k > j:
                        j+=1
                elif num + n1 + n2 > 0:
                    k-=1
                else:
                    j+=1
        
        return solution
        